src/Streamly/Internal/Data/Parser/ParserK/Types.hs

```diff
@@ -145,6 +149,19 @@ instance Monad m => Monad (Parser m a) where
                 Left err -> runParser (die err) z yieldk
         in runParser m inp yield1
 
+#if !(MIN_VERSION_base(4,13,0))
```

Comment:
Do we need this for base > 4.8 && < 4.13?

Reply:
Yes, `fail` is a method of `Monad` in those versions